From 4ea3c66a69236c535f4f80b0b20cbc58d980be70 Mon Sep 17 00:00:00 2001
From: mayi <3363523343@qq.com>
Date: Wed, 21 Oct 2026 18:34:14 +0800
Subject: [PATCH] feat(osdrv/atbm6461): add CV181x kernel and tool build
 entries

Select RHEA/SDIO and the vendor 24 MHz clock, add an out-of-tree Kbuild and cross-compile the original supplicant/CLI. Keep runtime installation and the standalone network manager out of this patch.
---
 extdrv/wireless/realtek/atbm6461/.config      |   6 +-
 extdrv/wireless/realtek/atbm6461/Kbuild       |  26 +
 extdrv/wireless/realtek/atbm6461/Makefile     | 469 +-----------------
 .../wireless/realtek/atbm6461/cv181x.config   |   9 +
 .../realtek/atbm6461/hal_apollo/svn_version.h |   1 +
 .../wireless/realtek/atbm6461/tools/Makefile  |  64 +--
 6 files changed, 76 insertions(+), 499 deletions(-)
 create mode 100644 extdrv/wireless/realtek/atbm6461/Kbuild
 create mode 100644 extdrv/wireless/realtek/atbm6461/cv181x.config

diff --git a/extdrv/wireless/realtek/atbm6461/.config b/extdrv/wireless/realtek/atbm6461/.config
index 901f0928d9543bb279b184a0759773dc1ad1980b..167a427fa6e1167e4a44baa9c894eca7c63e737b 100644
--- a/extdrv/wireless/realtek/atbm6461/.config
+++ b/extdrv/wireless/realtek/atbm6461/.config
@@ -4,8 +4,8 @@
 #
 CONFIG_ATBM_MENUCONFIG=y
 CONFIG_ATBM_WIRELESS=y
-CONFIG_ATBM6441=y
-# CONFIG_ATBM6461 is not set
+# CONFIG_ATBM6441 is not set
+CONFIG_ATBM6461=y
 # CONFIG_ATBM6463 is not set
 # CONFIG_ATBM_USB_BUS is not set
 CONFIG_ATBM_SDIO_BUS=y
@@ -37,4 +37,4 @@ CONFIG_ATBM_FUNC_MONITOR=y
 # CONFIG_ATBM_APOLLO_USE_GPIO_IRQ is not set
 CONFIG_ATBM_WIFIIF1_NAME="wlan%d"
 CONFIG_ATBM_WIFIIF2_NAME="p2p%d"
-CONFIG_ATBM_MODULE_NAME="atbm6441_wifi_sdio"
+CONFIG_ATBM_MODULE_NAME="atbm6461_wifi_sdio"
diff --git a/extdrv/wireless/realtek/atbm6461/Kbuild b/extdrv/wireless/realtek/atbm6461/Kbuild
new file mode 100644
index 0000000000000000000000000000000000000000..2789bee6d47f351e24993680828cfdceefb2bd94
--- /dev/null
+++ b/extdrv/wireless/realtek/atbm6461/Kbuild
@@ -0,0 +1,26 @@
+include $(src)/cv181x.config
+
+export CONFIG_ATBM_APOLLO := m
+export CONFIG_HERA := n
+export CONFIG_RHEA := y
+export CONFIG_EROS := n
+export SDIO_BUS := y
+export USB_BUS := n
+export SPI_BUS := n
+export MODULES_NAME := atbm6461_wifi_sdio
+export IF1NAME := wlan%d
+export IF2NAME := p2p%d
+export FW := fw.bin
+export LOAD_FW_H := $(CONFIG_ATBM_USE_FIRMWARE_H)
+export SKB_DEBUG := $(CONFIG_ATBM_FUNC_SKB_DEBUG)
+export MEM_DEBUG := $(CONFIG_ATBM_FUNC_MEM_DEBUG)
+export PMRELODDFW := n
+export EARLYSUSPEND := n
+export CONFIG_ATBM_SDIO_ATCMD
+export CONFIG_ATBM_APOLLO_DEBUGFS
+export CONFIG_ATBM_APOLLO_USE_GPIO_IRQ
+export ATBM_WIFI__EXT_CCFLAGS := -DATBM_WIFI_PLATFORM=30
+export NOSTDINC_FLAGS := -I$(src)/include \
+	-include $(src)/include/linux/compat-2.6.h -DCOMPAT_STATIC
+
+obj-m += hal_apollo/
diff --git a/extdrv/wireless/realtek/atbm6461/Makefile b/extdrv/wireless/realtek/atbm6461/Makefile
index 96f78a3390568c3c7ecfc42ae07ecfac84e4659e..b59255c22785deae84381e6eacc4783c6ceb7b8c 100644
--- a/extdrv/wireless/realtek/atbm6461/Makefile
+++ b/extdrv/wireless/realtek/atbm6461/Makefile
@@ -1,457 +1,30 @@
-#
-#makefile for build atbm_wifi.ko
-#
-VER_GIT=y
-DEBUG_MODE=y
-###############################################################################
-#
-# when release to customer ,the CUSTOMER_SUPPORT_USED must set to y!!!!!!!!!!!!!
-#
-###############################################################################
-CUSTOMER_SUPPORT_USED=y
-###############################################################################
-#PLATFORM_XUNWEI    		1
-#PLATFORM_SUN6I					2
-#PLATFORM_FRIENDLY			3
-#PLATFORM_SUN6I_64			4
-#PLATFORM_HI3798M				5
-#PLATFORM_HI3518E				6
-#PLATFORM_X86PC         7
-#PLATFORM_AMLOGIC				8
-#PLATFORM_AMLOGIC_905X	9
-#PLATFORM_ROCKCHIP      10
-#PLATFORM_MSTAR					11
-#PLATFORM_CDLINUX				12
-#PLATFORM_AMLOGIC_S805	13
-#PLATFORM_HIS_LINUX_3_4				14
-#PLATFORM_ROCKCHIP_3229				15
-#PLATFORM_ROCKCHIP_3229_ANDROID8		16
-#PLATFORM_HS_IPC				17
-#PLATFORM_SIGMASTAR                           18
-#PLATFORM_HI3516EV200                         19
-#PLATFORM_XUNWEI_2G                            20
-#PLATFORM_NVT98517       21
-#PLATFORM_AK_IPC       						22
-#PLATFORM_SUN8I       						23
-#PLATFORM_INGENICT31				24
-#PLATFORM_INGENICT41				25
-#PLATFORM_SUN50IWP1						26
-#PLATFORM_KALI                                  27
-export
-platform ?=PLATFORM_SUN50IWP1
-#Android
-#Linux
-sys ?= linux
-#arch:arm or arm64 or mips(NVT98517)
-arch ?= arm
-#export 
-#ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=$(platform)
+DRIVER_DIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+KERNEL_DIR ?= $(KERDIR)
+ARCH ?= arm
+CROSS_COMPILE ?= arm-linux-gnueabihf-
+WIFI_INSTALL_DIR ?= $(DRIVER_DIR)/driver_install
 
-ifeq ($(CUSTOMER_SUPPORT_USED),y)
-MAKEFILE_SUB ?= Makefile.build.customer
-else
-MAKEFILE_SUB ?= Makefile.build
-endif
-ifeq ($(PLATFORM_KALI),y)
-KERDIR:=/lib/modules/5.16.0-kali7-amd64/build/
-endif
-ifeq ($(platform),PLATFORM_NVT98517)
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-#KERDIR:=/home/xzq/svn_nvt_sdk/new_ipc_sdk/sdk_2.0/98517/6032i/NA51023_BSP_v1.1.1/linux-kernel/OUTPUT
-KERDIR:=/wifi_prj/staff/wangsiyao/baichuan/linux-kernel/OUTPUT/
-CROSS_COMPILE:=/wifi_prj/staff/wangsiyao/baichuan/mipsel-24kec-linux-uclibc-4.9-2017.07/usr/bin/mipsel-24kec-linux-uclibc-
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=21
-arch = mips
-endif
-
-
-ifeq ($(KERNELRELEASE),)
-export
-ATBM_BUILD_IN_KERNEL=n
-export DRIVER_PATH ?= $(shell pwd)
-ifeq ($(platform),PLATFORM_HS_IPC)
-KERDIR:=/wifi_prj/staff/jilechang/HS_IPC_1115/kernel
-CROSS_COMPILE:=/wifi_prj/staff/jilechang/HS_IPC_1115/opt/arm-anykav200-crosstool/usr/bin/arm-anykav200-linux-uclibcgnueabi-
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=17
-arch = arm
-endif
-
-ifeq ($(platform),PLATFORM_ROCKCHIP_3229_ANDROID8)
-ifeq ($(sys),Android)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/rk3328_box_oreo_release_v1.0.0_20180319/kernel
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/rk3328_box_oreo_release_v1.0.0_20180319/prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9/bin/aarch64-linux-android-
-else
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-KERDIR:=/wifi_prj/wuping/project/linux/iTop4412_Kernel_3.0/
-CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=10
-arch = arm64
-endif
-
-ifeq ($(platform),PLATFORM_ROCKCHIP_3229)
-ifeq ($(sys),Android)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/rockchip_3229_kernel/kernel/
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/rockchip_3229_kernel/prebuilts/gcc/linux-x86/arm/arm-eabi-4.6/bin/arm-eabi-
-else
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-KERDIR:=/wifi_prj/wuping/project/linux/iTop4412_Kernel_3.0/
-CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=10
-endif
-ifeq ($(platform),PLATFORM_HIS_LINUX_3_4)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/his_liux3_4/linux-3.4.y/
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/his_liux3_4/arm-hisiv300-linux/bin/arm-hisiv300-linux-uclibcgnueabi-
-arch = arm
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=14
-endif
-ifeq ($(platform),PLATFORM_MSTAR)
-ifeq ($(sys),Linux)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/mstar/linux3.18_i3/
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/mstar/arm-linux-gnueabihf-4.8.3-201404/bin/arm-linux-gnueabihf-
-else
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-KERDIR:=/wifi_prj/wuping/project/linux/iTop4412_Kernel_3.0/
-CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-endif
-export
-arch = arm
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=11
-endif
-
-ifeq ($(platform),PLATFORM_SIGMASTAR)
-ifeq ($(sys),Linux)
-KERDIR:=/wifi_prj/staff/futianhuang/sigma_star/linux-3.18/
-CROSS_COMPILE:=/wifi_prj/staff/futianhuang/sigma_star/arm-linux-gnueabihf-4.8.3-201404/bin/arm-linux-gnueabihf-
-else
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-#KERDIR:=/wifi_prj/wuping/project/linux/iTop4412_Kernel_3.0/
-#CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-endif
-export
-arch = arm
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=18
-endif
-
-
-ifeq ($(platform),PLATFORM_ROCKCHIP)
-ifeq ($(sys),Android)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/rockchip_new/kernel1/
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/rockchip_new/prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9/bin/aarch64-linux-android-
-else
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-KERDIR:=/wifi_prj/wuping/project/linux/iTop4412_Kernel_3.0/
-CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-endif
-export
-arch = arm64
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=10 
+ifeq ($(filter arm arm64,$(ARCH)),)
+$(error ATBM6461 BF01 requires ARCH=arm or ARCH=arm64)
 endif
 
-#
-#xunwei platform params
-#
-ifeq ($(platform),PLATFORM_XUNWEI)
-ifeq ($(sys),Android)
-#KERDIR:=/wifi_prj/zhouzhanchao/android4_4_2/iTop4412_Kernel_3.0/
-KERDIR:=/wifi_prj/wuping/project/android4_4_2/iTop4412_Kernel_3.0
-CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-else
-#KERDIR:=/wifi_prj/wuping/project/linux/iTop4412_Kernel_3.0/
-KERDIR:=/wifi_prj/staff/panxuqiang/wifi_prj/iTop4412_Kernel_3.0/
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-#KERDIR:=/wifi_prj/staff/jilechang/XUNWEI/iTop4412_Kernel_3.0/
-#CROSS_COMPILE:=/wifi_prj/staff/jilechang/XUNWEI/arm-2009q3/bin/arm-none-linux-gnueabi-
+.PHONY: all modules tools tools_install install clean
 
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=1
-endif
-ifeq ($(platform),PLATFORM_XUNWEI_2G)
-ifeq ($(sys),Android)
-#KERDIR:=/wifi_prj/zhouzhanchao/android4_4_2/iTop4412_Kernel_3.0/
-KERDIR:=/wifi_prj/wuping/project/android4_4_2/iTop4412_Kernel_3.0
-CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-else
-#KERDIR:=/wifi_prj/wuping/project/linux/iTop4412_Kernel_3.0/
-#KERDIR:=/wifi_prj/staff/panxuqiang/wifi_prj/iTop4412_Kernel_3.0/
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/iTop4412_Kernel_3.0/
-#CROSS_COMPILE:=/usr/local/arm/arm-2009q3/bin/arm-none-linux-gnueabi-
-KERDIR:=/wifi_prj/staff/jilechang/XUNWEI/iTop4412_Kernel_3.0/
-CROSS_COMPILE:=/wifi_prj/staff/jilechang/XUNWEI/arm-2009q3/bin/arm-none-linux-gnueabi-
+all: modules tools
 
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=20
-endif
+modules:
+	test -n "$(KERNEL_DIR)"
+	test -s $(KERNEL_DIR)/.config
+	test -s $(KERNEL_DIR)/Module.symvers
+	$(MAKE) -C $(KERNEL_DIR) M=$(DRIVER_DIR) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) modules
 
-#
-#allwinner 32 platform params
-#
-ifeq ($(platform),PLATFORM_SUN6I)
-ifeq ($(sys),Android)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/android4_4_SIN/Source/lichee/linux-3.3
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/android4_4_SIN/Source/lichee/brandy/gcc-linaro/bin/arm-linux-gnueabi-
-else
-KERDIR:=/wifi_prj/staff/wangzhiguang/Linux_sun6i/lichee/linux-3.3/
-#KERDIR:=/wifi_prj/staff/songningning/lichee_sun6i/linux-3.3/
-#CROSS_COMPILE:=/wifi_prj/staff/wangzhiguang/Linux_sun6i/lichee/buildroot/output/external-toolchain/bin/arm-linux-gnueabi-
-CROSS_COMPILE:= /wifi_prj/staff/mengxuehong/mengxuehong/arm-2014.05/bin/arm-none-linux-gnueabi-
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=2
-endif
-
-#
-#allwinner 32 platform params
-#
-ifeq ($(platform),PLATFORM_HUALAI)
-##################wpa3 cfg80211 kernel used songningning/lichee_sun6i/linux-3.3/#####################
-KERDIR:=/wifi_prj/staff/muqing/hualai/kernel-4.4.94/
-#CROSS_COMPILE:=/wifi_prj/staff/muqing/hualai/mips-gcc720-glibc229-r5.1.4/bin/mips-linux-gnu-
-CROSS_COMPILE:=/wifi_prj/staff/muqing/hualai/mips-gcc720-uclibc0.9.33.2-r5.1.4/bin/mips-linux-uclibc-
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=30
-endif
-
-#
-#allwinner 64 platform params
-#
-ifeq ($(platform),PLATFORM_SUN6I_64)
-ifeq ($(sys),Android)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/android6_0_SIN/cqa64_android_v6.0/lichee/linux-3.10/
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/android6_0_SIN/cqa64_android_v6.0/lichee/out/sun50iw1p1/android/common/buildroot/external-toolchain/bin/aarch64-linux-gnu-
-else
-KERDIR:=/wifi_prj/staff/panxuqiang/64bi_driver/cqa64_linux_qt5.3.2/lichee/linux-3.10/
-CROSS_COMPILE:=/wifi_prj/staff/panxuqiang/64bi_driver/cqa64_linux_qt5.3.2/lichee/brandy/armv8_toolchain/gcc-linaro-aarch64-linux-gnu-4.9-2014.09_linux/bin/aarch64-linux-gnu-
-endif
-arch:=arm64
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=4
-endif
+tools tools_install:
+	$(MAKE) -C $(DRIVER_DIR) -f tools/Makefile CROSS_COMPILE=$(CROSS_COMPILE) WIFI_INSTALL_DIR=$(WIFI_INSTALL_DIR) tools_install
 
-#
-#HI3798M platform params
-#
-ifeq ($(platform),PLATFORM_HI3798M)
-ifeq ($(sys),Android)
-KERDIR:=/ssd-home/zhouzhanchao/Hi3798M-60-2/Hi3798M-60/out/target/product/Hi3798MV100/obj/KERNEL_OBJ/
-CROSS_COMPILE:=/opt/hisi-linux/x86-arm/arm-hisiv200-linux/target/bin/arm-hisiv200-linux-
-else
-#KERDIR:=/wifi_prj/staff/panxuqiang/64bi_driver/cqa64_linux_qt5.3.2/lichee/linux-3.10/
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=5
-endif
-
-ifeq ($(platform),PLATFORM_PCX86)
-KERDIR:=/kernel/linux-lts-utopic-3.16.0/
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=7
-arch:=x86
-MAKEFILE_SUB = Makefile.build.local
-endif
-ifeq ($(platform),PLATFORM_AMLOGIC)
-ifeq ($(sys),Android)
-#KERDIR:=/wifi_prj/staff/zhouzhanchao/amlogic/release_s905_l/out/target/product/p200/obj/KERNEL_OBJ/
-KERDIR:=/wifi_prj/staff/mengxuehong/s905l/S905L/out/target/product/p201_iptv/obj/KERNEL_OBJ/
-CROSS_COMPILE:=/ssd-home/mengxuehong/buildTool1/gcc-linaro-aarch64-linux-gnu-4.9-2014.09_linux/bin/aarch64-linux-gnu-
-else
-#KERDIR:=/wifi_prj/staff/panxuqiang/64bi_driver/cqa64_linux_qt5.3.2/lichee/linux-3.10/
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=8
-arch:=arm64
-endif
-
-
-ifeq ($(platform),PLATFORM_AMLOGIC_S805)
-ifeq ($(sys),Android)
-KERDIR:=/wifi_prj/staff/zhouzhanchao/s805_env/KERNEL_OBJ/
-CROSS_COMPILE:=/wifi_prj/staff/zhouzhanchao/s805_env/opt/gcc-linaro-arm-linux-gnueabihf/bin/arm-linux-gnueabihf-
-else
-#KERDIR:=/wifi_prj/staff/panxuqiang/64bi_driver/cqa64_linux_qt5.3.2/lichee/linux-3.10/
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=8
-arch:=arm
-endif
+install: all
+	install -d $(WIFI_INSTALL_DIR)
+	install -m 0644 $(DRIVER_DIR)/hal_apollo/atbm6461_wifi_sdio.ko $(WIFI_INSTALL_DIR)/
 
-ifeq ($(platform),PLATFORM_AMLOGIC_905X)
-ifeq ($(sys),Android)
-KERDIR=/wifi_prj/staff/mengxuehong/mengxuehong/aml0804/n-amlogic_0804/out/target/product/p281/obj/KERNEL_OBJ
-#KERDIR:=/wifi_prj/staff/mengxuehong/mengxuehong/amlogic_sdk/release_s905_l/out/target/product/p201/obj/KERNEL_OBJ/
-CROSS_COMPILE:=/wifi_prj/staff/mengxuehong/mengxuehong/amlogic_sdk/buildTool/gcc-linaro-aarch64-linux-gnu-4.9-2014.09_linux/bin/aarch64-linux-gnu-
-else
-#KERDIR:=/wifi_prj/staff/panxuqiang/64bi_driver/cqa64_linux_qt5.3.2/lichee/linux-3.10/
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=9
-arch:=arm64
-endif
-
-ifeq ($(platform),PLATFORM_HI3516EV200)
-KERDIR:=/wifi_prj/staff/panxuqiang/wifi_prj/Hi3516EV200_SDK_V1.0.0.2/osdrv/opensource/kernel/linux-4.9.y
-CROSS_COMPILE:=/opt/hisi-linux/x86-arm/arm-himix100-linux/bin/arm-himix100-linux-
-export 
-arch = arm
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=19
-#ATBM_WIFI__EXT_CCFLAGS += -mcpu=cortex-a7 -mfloat-abi=softfp -mfpu=neon-vfpv4 -fno-aggressive-loop-optimizations
-endif
-
-ifeq ($(platform),PLATFORM_AK_IPC)
-KERDIR := ${KSRC)
-KVER := ${shell uname -r)
-CROSS_COMPILE:=arm-anykav500-linux-uclibcgnueabi-
-arch = arm
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=22
-#ATBM_WIFI__EXT_CCFLAGS += -mcpu=cortex-a7 -mfloat-abi=softfp -mfpu=neon-vfpv4 -fno-aggressive-loop-optimizations
-endif
-
-#
-#allwinner a33 platform params
-#
-ifeq ($(platform),PLATFORM_SUN8I)
-ifeq ($(sys),Android)
-else
-KERDIR:=/wifi_prj/staff/songningning/lichee/linux-3.4/
-CROSS_COMPILE:=/wifi_prj/staff/songningning/lichee/out/sun8iw5p1/linux/common/buildroot/external-toolchain/bin/arm-linux-gnueabi-
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=23
-endif
-
-#
-#allwinner H64 platform params
-#
-ifeq ($(platform),PLATFORM_SUN50IWP1)
-arch:=arm64
-ifeq ($(sys),Android)
-KERDIR:=/usr/lchome/zhaoledong/Android6.0/cqa64_android_v6.0/lichee/linux-3.10
-CROSS_COMPILE:=/wifi_prj/staff/songningning/cqa64_ubuntu18_64bit_bv3s/out/sun50iw1p1/ubuntu18/common/buildroot/external-toolchain/bin/aarch64-linux-gnu-
-#CROSS_COMPILE:=/usr/lchome/zhaoledong/Android6.0/cqa64_android_v6.0/lichee/brandy/armv8_toolchain/gcc-linaro-aarch64-linux-gnu-4.9-2014.09_linux/bin/aarch64-linux-gnu-
-else
-KERDIR:=/wifi_prj/staff/songningning/cqa64_ubuntu18_64bit_bv3s/linux-3.10
-CROSS_COMPILE:=/wifi_prj/staff/songningning/cqa64_ubuntu18_64bit_bv3s/out/sun50iw1p1/ubuntu18/common/buildroot/external-toolchain/bin/aarch64-linux-gnu-
-endif
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=26
-endif
-
-ifeq ($(platform),PLATFORM_CDLINUX)
-KERDIR:=/lib/modules/3.2.11_cdl0/build
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=11
-endif
-
-ifeq ($(platform),PLATFORM_PCX86)
-all:install
-
-install:
-	@echo "make PLATFORM_PCX86"
-	$(MAKE) all -f Makefile.build.local KDIR=$(KERDIR)
-clean:
-	$(MAKE) -f Makefile.build.local KDIR=$(KERDIR) clean
-else
-all:install tools_install strip
-
-ifeq ($(VER_GIT),y)
-GIT_HEX_VER := $(shell git rev-parse --short HEAD 2>NUL)
-LMAC_ADD_TIME := $(shell git log -1 --format="%cd" --date=format:"%Y%m%d%H")
-ATBM_SVN_VERSION = $(GIT_HEX_VER)
-ATBM_SVN_TIME = $(LMAC_ADD_TIME)
-else
-ATBM_SVN_VERSION := 0000
-ATBM_SVN_TIME := 0000
-endif
-getsvn:
-	@echo "**************************************"
-ifeq ($(DEBUG_MODE),y)
-	@echo "#define DRIVER_VER 0x"$(ATBM_SVN_VERSION) >hal_apollo/svn_version.h
-	@echo "#define DRIVER_VER_TIME " $(ATBM_SVN_TIME) >>hal_apollo/svn_version.h
-	@echo "svn version:" $(ATBM_SVN_VERSION)
-endif
-
-install:getsvn
-	@echo "make PLATFORM_CROSS=$(platform)"
-	$(MAKE) all -f $(MAKEFILE_SUB) ARCH=$(arch)  CROSS_COMPILE=$(CROSS_COMPILE) KDIR=$(KERDIR) SYS=$(sys) PLAT=$(platform) -j8
 clean:
-	$(MAKE) -f $(MAKEFILE_SUB) KDIR=$(KERDIR) ARCH=$(arch) clean
-strip:	
-	$(MAKE) -f $(MAKEFILE_SUB) KDIR=$(KERDIR) ARCH=$(arch) SYS=$(sys) PLAT=$(platform) strip
-get_ver:
-	$(MAKE) -f $(MAKEFILE_SUB) KDIR=$(KERDIR) ARCH=$(arch) SYS=$(sys) PLAT=$(platform) get_ver
-tools_install:
-	$(MAKE) tools_install -f $(MAKEFILE_SUB) ARCH=$(arch)  CROSS_COMPILE=$(CROSS_COMPILE) KDIR=$(KERDIR) SYS=$(sys) PLAT=$(platform) -j8
-
-power_install:
-	$(MAKE) power_install -f $(MAKEFILE_SUB) ARCH=$(arch)  CROSS_COMPILE=$(CROSS_COMPILE) KDIR=$(KERDIR) SYS=$(sys) PLAT=$(platform) -j8
-
-buid_config:
-	$(MAKE) -C atbm_kconf mconf -f Makefile
-menuconfig:buid_config
-	@./atbm_kconf/mconf ./atbm_kconf/Kconfig
-endif
-else
-ifeq ($(platform),PLATFORM_XUNWEI)
-export 
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=1
-endif
-ifeq ($(platform),PLATFORM_SUN6I)
-export 
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=2
-endif
-ifeq ($(platform),PLATFORM_SUN6I_64)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=4
-endif
-ifeq ($(platform),PLATFORM_HI3798M)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=5
-endif
-ifeq ($(platform),PLATFORM_AMLOGIC)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=8
-endif
-ifeq ($(platform),PLATFORM_AMLOGICi_905X)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=9
-endif
-ifeq ($(platform),PLATFORM_ROCKCHIP)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=10
-endif
-ifeq ($(platform),PLATFORM_MSTAR)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=11
-endif
-ifeq ($(platform),PLATFORM_CDLINUX)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=12
-endif
-ifeq ($(platform),PLATFORM_AMLOGIC_S805)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=13
-endif
-ifeq ($(platform),PLATFORM_ROCKCHIP_3229)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=10
-endif
-ifeq ($(platform),PLATFORM_NVT98517)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=14
-endif
-ifeq ($(platform),PLATFORM_SUN50IWP1)
-export
-ATBM_WIFI__EXT_CCFLAGS = -DATBM_WIFI_PLATFORM=26
-endif
-export 
-include $(src)/Makefile.build.kernel
-endif
-
-
-
+	$(MAKE) -C $(KERNEL_DIR) M=$(DRIVER_DIR) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) clean
+	$(MAKE) -C $(DRIVER_DIR) -f tools/Makefile WIFI_INSTALL_DIR=$(WIFI_INSTALL_DIR) clean
diff --git a/extdrv/wireless/realtek/atbm6461/cv181x.config b/extdrv/wireless/realtek/atbm6461/cv181x.config
new file mode 100644
index 0000000000000000000000000000000000000000..386943719c675523f8285ca0830183ae88d46ece
--- /dev/null
+++ b/extdrv/wireless/realtek/atbm6461/cv181x.config
@@ -0,0 +1,9 @@
+CONFIG_ATBM6461=y
+CONFIG_ATBM_SDIO_BUS=y
+CONFIG_ATBM_USE_FIRMWARE_BIN=y
+CONFIG_ATBM_USE_FIRMWARE_H=n
+CONFIG_ATBM_FUNC_SKB_DEBUG=n
+CONFIG_ATBM_FUNC_MEM_DEBUG=n
+CONFIG_ATBM_SDIO_ATCMD=n
+CONFIG_ATBM_APOLLO_DEBUGFS=n
+CONFIG_ATBM_APOLLO_USE_GPIO_IRQ=n
diff --git a/extdrv/wireless/realtek/atbm6461/hal_apollo/svn_version.h b/extdrv/wireless/realtek/atbm6461/hal_apollo/svn_version.h
index 28c8a115e1db08b2d9b6ae5eedec004d962e66b5..863084e7e616717fc9a5b8ea015c935787c44630 100644
--- a/extdrv/wireless/realtek/atbm6461/hal_apollo/svn_version.h
+++ b/extdrv/wireless/realtek/atbm6461/hal_apollo/svn_version.h
@@ -1 +1,2 @@
 #define DRIVER_VER  111
+#define DRIVER_VER_TIME 20260408
diff --git a/extdrv/wireless/realtek/atbm6461/tools/Makefile b/extdrv/wireless/realtek/atbm6461/tools/Makefile
index d071af38ffa848d738a4da06086763db3fd06ecb..cb041199f3e2e6120b3d2a0e1d996354096861b1 100644
--- a/extdrv/wireless/realtek/atbm6461/tools/Makefile
+++ b/extdrv/wireless/realtek/atbm6461/tools/Makefile
@@ -1,56 +1,24 @@
--include ../.config
+include cv181x.config
 
-TOOL_NAME:=atbm_iot_supplicant
-TOOL_DEMO_NAME:=atbm_iot_supplicant_demo
-TOOL_CLI_NAME:=atbm_iot_cli
-#MODULE_NAME:=power_ctrl
+WIFI_INSTALL_DIR ?= driver_install
+TOOL_CFLAGS := -Itools -Ihal_apollo -DARES_B=10 -DHERA=11 -DRHEA=12 -DEROS=13 -DPROJ_TYPE=RHEA
 
-NOSTDINC_FLAGS :=
-
-#$(MODULE_NAME)-y = power_func.o
+ifeq ($(CONFIG_ATBM_SDIO_ATCMD),y)
+TOOL_CFLAGS += -DCONFIG_ATBM_SDIO_ATCMD
+endif
 
-#obj-m = $(MODULE_NAME).o
+.PHONY: tools_install clean
 
-tool_cflags += -I./ -I./hal_apollo/ -lpthread
+tools_install: $(WIFI_INSTALL_DIR)/atbm_iot_supplicant $(WIFI_INSTALL_DIR)/atbm_iot_cli
 
-ifeq ($(CONFIG_ATBM_SDIO_ATCMD),y)
-tool_cflags += -DCONFIG_ATBM_SDIO_ATCMD
-endif
+$(WIFI_INSTALL_DIR):
+	install -d $@
 
-CONFIG_ARESB   ?= $(CONFIG_ATBM603x)
-CONFIG_HERA    ?= $(CONFIG_ATBM6441)
-CONFIG_RHEA    ?= $(CONFIG_ATBM6461)
-CONFIG_EROS    ?= $(CONFIG_ATBM6463)
-tool_cflags += -DARES_B=10 -DHERA=11 -DRHEA=12 -DEROS=13
+$(WIFI_INSTALL_DIR)/atbm_iot_supplicant: tools/tools.c tools/atbm_tool.h hal_apollo/atbm_ioctl.h cv181x.config tools/Makefile | $(WIFI_INSTALL_DIR)
+	$(CROSS_COMPILE)gcc $(CPPFLAGS) $(CFLAGS) $(TOOL_CFLAGS) $< $(LDFLAGS) -pthread -lrt -o $@
 
-ifeq ($(CONFIG_ARESB),y)
-tool_cflags += -DPROJ_TYPE=ARES_B
-endif
-ifeq ($(CONFIG_HERA),y)
-tool_cflags += -DPROJ_TYPE=HERA
-endif
-ifeq ($(CONFIG_RHEA),y)
-tool_cflags += -DPROJ_TYPE=RHEA
-endif
-ifeq ($(CONFIG_EROS),y)
-tool_cflags += -DPROJ_TYPE=EROS
-endif
+$(WIFI_INSTALL_DIR)/atbm_iot_cli: tools/tools_cli.c tools/atbm_tool.h cv181x.config tools/Makefile | $(WIFI_INSTALL_DIR)
+	$(CROSS_COMPILE)gcc $(CPPFLAGS) $(CFLAGS) $(TOOL_CFLAGS) $< $(LDFLAGS) -o $@
 
-tools_install:
-	$(CROSS_COMPILE)gcc tools/tools.c $(tool_cflags) -o $(TOOL_NAME) -lrt
-	$(CROSS_COMPILE)gcc tools/tools.c $(tool_cflags) -o $(TOOL_DEMO_NAME) -lrt
-#	$(CROSS_COMPILE)gcc tools/tools.c $(tool_cflags) -DDEMO_TCP_SEND -o $(TOOL_DEMO_NAME)
-	$(CROSS_COMPILE)gcc tools/tools_cli.c $(tool_cflags) -o $(TOOL_CLI_NAME)
-	cp $(TOOL_NAME) driver_install/
-	cp $(TOOL_DEMO_NAME) driver_install/
-	cp $(TOOL_CLI_NAME) driver_install/
-	
-#power_install: power_clean
-#	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KDIR) M=$(shell pwd)/tools/ modules -j8
-#	cp tools/*.ko driver_install/
-	
-#power_clean:
-#	rm -rf tools/*.ko
-#	rm -rf tools/*.o
-#	rm -rf tools/*.mod.c
-#	rm -rf tools/*.cmd
+clean:
+	$(RM) $(WIFI_INSTALL_DIR)/atbm_iot_supplicant $(WIFI_INSTALL_DIR)/atbm_iot_cli
